src/datastore/memory-store.ts

```diff
@@ -303,6 +304,18 @@ export class MemoryDataStore extends (EventEmitter as { new (): DataStoreEventEm
     throw new Error('not yet implemented');
   }
 
+  searchHash(args: {
+    hash: string;
+  }): Promise<{ found: false } | { found: true; result: DbSearchResult }> {
```

Comment:
Loads of these `{found : false } | {found: true, result: X}` types. Could make a generic:

```
type FoundOrNot<T> = { found: false } | { found: true; result: T }
```

Reply:
Great idea 👍 
That change will touch a bunch of files, will do in a follow up PR